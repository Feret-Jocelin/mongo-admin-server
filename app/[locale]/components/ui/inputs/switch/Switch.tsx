import styles from './switch.module.scss';

import GenericButton from "@components/ui/button/GenericButton";

interface SwitchProps {
  selected: string;
  options: string[];
  onChange: (value: string) => void;
}

const Switch = ({ selected, options, onChange }: SwitchProps) => {
  return (
    <div className={styles['switch-container']}>
      {options.map((option, index) => (
        <GenericButton
          key={index}
          transparent={selected !== option}
          variant={selected === option ? 'info' : undefined}
          outline
          center
          onClick={() => onChange(option)}
        >
          {option}
        </GenericButton>
      ))}
    </div>
  );
}

export default Switch;