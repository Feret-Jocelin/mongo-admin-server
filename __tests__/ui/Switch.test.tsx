/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import Switch from '@/app/[locale]/components/ui/inputs/switch/Switch';

describe('Switch', () => {
    it('renders', () => {
      const handleChange = jest.fn();
      const { container } = render(
        <Switch 
          selected='table'
          options={['table', 'json']}
          onChange={handleChange} 
        />
      );
      expect(container).toMatchSnapshot();
    })
  })