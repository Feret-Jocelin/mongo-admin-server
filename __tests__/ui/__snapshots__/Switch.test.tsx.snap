// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders 1`] = `
<div>
  <div
    class="switch-container"
  >
    <button
      class="button full outline center info"
      type="button"
    >
      table
    </button>
    <button
      class="button full outline center transparent"
      type="button"
    >
      json
    </button>
  </div>
</div>
`;
